.cbmc-batch/jobs/Makefile.common

```diff
@@ -124,90 +134,103 @@ setup_dependencies:
+	echo "Not implementing abstractions" | tee $(call LOG_FROM_ENTRY, $@) ; exit $${PIPESTATUS[0]}
 else
 	$(GOTO_CC) --function $(ENTRY) $(ABSTRACTIONS) $< $(INC) $(DEFINES) -o $@ \
-		2>&1 | tee $(ENTRY)2.log ; exit $${PIPESTATUS[0]}
+		2>&1 | tee $(call LOG_FROM_ENTRY, $@); exit $${PIPESTATUS[0]}
 endif
 
 
 ## Temporarily skipped steps
 
 # Simplify and constant propagation may benefit from unwinding first
-$(ENTRY)3.goto: $(ENTRY)2.goto
+$(ENTRY_GOTO)3.goto: $(ENTRY_GOTO)2.goto
 ifeq ($(UNWIND_GOTO), 1)
 	$(GOTO_INSTRUMENT) $(UNWINDING) $< $@ \
-		2>&1 | tee $(ENTRY)3.log ; exit $${PIPESTATUS[0]}
+		2>&1 | tee $(call LOG_FROM_ENTRY, $@) ; exit $${PIPESTATUS[0]}
 else
 	cp $< $@
-	echo "Not unwinding goto program" | tee $(ENTRY)3.log ; exit $${PIPESTATUS[0]}
+	echo "Not unwinding goto program" | tee $(call LOG_FROM_ENTRY, $@) ; exit $${PIPESTATUS[0]}
 endif
 
 # Skip simplify (and hence generate-function-body) until missing source locations debugged
-$(ENTRY)4.goto: $(ENTRY)3.goto
+$(ENTRY_GOTO)4.goto: $(ENTRY_GOTO)3.goto
 ifeq ($(SIMPLIFY), 1)
 	$(GOTO_INSTRUMENT) --generate-function-body '.*' $< $@ \
-		2>&1 | tee $(ENTRY)4.log ; exit $${PIPESTATUS[0]}
+		2>&1 | tee$(call LOG_FROM_ENTRY, $@); exit $${PIPESTATUS[0]}
 else
 	cp $< $@
-	echo "Not generating-function-bodies in goto program" | tee $(ENTRY)4.log ; exit $${PIPESTATUS[0]}
+	echo "Not generating-function-bodies in goto program" | tee $(call LOG_FROM_ENTRY, $@) ; exit $${PIPESTATUS[0]}
 endif
 
 # Skip simplify (and hence generate-function-body) until missing source locations debugged
-$(ENTRY)5.goto: $(ENTRY)4.goto
+$(ENTRY_GOTO)5.goto: $(ENTRY_GOTO)4.goto
 ifeq ($(SIMPLIFY), 1)
 	$(GOTO_ANALYZER) --simplify $@ $< \
-		2>&1 | tee $(ENTRY)5.log ; exit $${PIPESTATUS[0]}
+		2>&1 | tee $(call LOG_FROM_ENTRY, $@); exit $${PIPESTATUS[0]}
 else
 	cp $< $@
-	echo "Not simplfying goto program" | tee $(ENTRY)5.log ; exit $${PIPESTATUS[0]}
+	echo "Not simplfying goto program" | tee $(call LOG_FROM_ENTRY, $@) ; exit $${PIPESTATUS[0]}
 endif
 
 # Simplify the goto program by removing any unused function bodies
-$(ENTRY)6.goto: $(ENTRY)5.goto
+$(ENTRY_GOTO)6.goto: $(ENTRY_GOTO)5.goto
 	$(GOTO_INSTRUMENT) --drop-unused-functions $< $@ \
-		2>&1 | tee $(ENTRY)6.log ; exit $${PIPESTATUS[0]}
+		2>&1 | tee $(call LOG_FROM_ENTRY, $@) ; exit $${PIPESTATUS[0]}
 
 # Simplify the goto program by slicing away initializations of unused
 # global variables
-$(ENTRY)7.goto: $(ENTRY)6.goto
+$(ENTRY_GOTO)7.goto: $(ENTRY_GOTO)6.goto
 	$(GOTO_INSTRUMENT) --slice-global-inits $< $@ \
-		2>&1 | tee $(ENTRY)7.log ; exit $${PIPESTATUS[0]}
+		2>&1 | tee $(call LOG_FROM_ENTRY, $@) ; exit $${PIPESTATUS[0]}
+
+$(ENTRY_GOTO).goto: $(ENTRY_GOTO)7.goto
+	cp $< $@
 
-$(ENTRY).goto: $(ENTRY)7.goto
+# For now, CBMC batch expects the goto in the same folder.  I could update the
+# .yaml file for all 160 proofs, or I could copy it, and have one goto file
+# in the main directory.  
+$(ENTRY).goto: $(ENTRY_GOTO).goto
```

Comment:
Just to confirm what is going on here: you're building all of the dependent goto object files out-of-tree, linking them to the harness goto file (also out-of-tree), then copying that final linked goto back into the source tree so that the proofs continue to just work?

I didn't initially know what `it` was referring to, I recommend clarifying the comment:
```make
# CBMC batch expects the goto binary to be in the same folder as the
# Makefile. Therefore copy the goto binary back to this directory
# from the out-of-tree build.
```

Reply:
Not quite.  We do have the out-of-source builds, in that all `.goto` files are now build under the proof directory.  However, I had tried to do a second level of cleanup, and move it so that all files would go in subdirectories of the proof itself called `logs` and `gotos`.

The `.yaml` file for the CI system expects the final `.goto` to be in the same folder as the proof itself.  So I can either update all `.yaml` files, or just copy the one `.goto` file.  But we're still building a separate build per proof.  I'll update the comment.

Reply:
Makes sense, thanks for the clarification